### ISSUE TITLE
Feature request: Event improvements

### ISSUE THREAD

**Issue by wootapa** [nextapps-de/winbox#46]
- It would be nice with a simple api to subscribe, unsubscribe for any published events. Maybe be adding methods such as `on`, `once` and `off` so clients can add multiple handlers to the same event, listen only once and stop listening at any time. It will also probably be easier for you to publish existing and new events later on as you only need to provide a list of whats observable.
- It might be considered another issue, but events that occur before and after is very convenient such as `before:move` and `after:move`,  `before:resize`, `after:resize` etc. I client could then for example return a promise when something is complete, like:

```javascript
myMoveFunction(x, y) {
  return new Promise((resolve, reject) => {
    winboxInstance.once('after:move', _ => resolve()); // <-- Finished moving
    winboxInstance.move(x, y);
  });
}
```
